core/extraction/towns_fund.py

```diff
@@ -437,7 +437,7 @@ def extract_funding_comments(df_input: pd.DataFrame, project_lookup: dict) -> pd
 
     for idx in range(len(project_lookup)):
         line_idx = 28 * idx
-        current_project = df_input.iloc[line_idx, 0].split(": ")[1]
+        current_project = ": ".join(df_input.iloc[line_idx, 0].split(": ")[1:])  # omit the "Project X: " prefix
```

Comment:
is this testable?

Reply:
Is `:` treated as a special character by the library or something? could we encode it differently or do a regex on it? Removing it like this may have unintended consequences.

Reply:
none of this is unit tested.. it's a bigger job to think about than this fix I think

Reply:
It's because the cell is always 
```
Project X: A city or something
```
but this SS was:
```
Project X: A city: something else
```

so it was originally just grabbing the "A city" bit, without the end, and that didnt match the project ID lookup table so it all fell over. This should work regardless of how many colons you have it in now

Reply:
Could we extract the logic to a function and unit test that?

Reply:
I could extract that line into it's own function and unit test that individual line's logic if you wanted but as I said, I think its small fry in comparison to all this un(unit)tested logic